### DISCUSSION TITLE
shouldFocusError not working on a custom input component v7.0

### DISCUSSION THREAD

**Discussion by nani1008** [react-hook-form/react-hook-form#4842] [Q&A] [answered]
So I have a custom input component but when I have error the input field is not focused, when I updated from 6.0 to 7.0 this happens. 

Here is my custom component 

> ### <input
>             
>             className="form-control"
>             ref={register}
>             {...inputProps}
>           />

          
This is when im using my custom component 

  

>                  <Input
>                     title="name"
>                     placeholder="name"
>                     {...register("name", {
>                       required: true,
>                     })}
>                   />

**Comment by bluebill1049** [comment, marked as answer]
check your ref name, you may need to assign ref to inputRef.

**Reply by nani1008** [reply]
@bluebill1049 what do you mean by this? I tried assigning a ref into register still not working. please help.

**Reply by nani1008** [reply]
please help me i dont get what you are saying to assign ref into inputRef can you provide simple code? thank you.

**Reply by bluebill1049** [reply]
can yo share a simple codesandbox with the problem? i will adjust the problem

**Reply by nani1008** [reply]
@bluebill1049 https://codesandbox.io/s/restless-tree-3fnrb?file=/src/App.js here I want the firstName to be focused if I clicked the submit button since it is required which will give an error and this is not happening on the v7.0. It is a custom input component but if I did not use custom component it is working. How can I solve this? Thank you!

**Reply by bluebill1049** [reply]
here you go: https://codesandbox.io/s/inspiring-solomon-bb93r?file=/src/CustomInput.js read the doc on get started section, it may help you a bit more there.

**Reply by nani1008** [reply]
Got it, did not see that I directly read the docs from the API so my bad. Thank you for your help.
